### PR TITLE
RFG "search again" continues search where it left previously

### PR DESCRIPTION
**Pull request by u-an-i**

### CODE REVIEW

File: src/search/airportsearch.cpp [hunk 1]
```diff
@@ -403,7 +403,6 @@ void AirportSearch::saveState()
   atools::gui::WidgetState widgetState(lnm::SEARCHTAB_AIRPORT_WIDGET);
   widgetState.save(airportSearchWidgets);
   saveViewState(viewStateDistSearch);
-  // TODO: make sure RFG state (min, max, predefined) is saved and restored.
 }
```
Comment (u-an-i) [inline src/search/airportsearch.cpp:406]: restore does not yet work for me, I haven't found the reason. Perhaps the min/max setting ?

Comment (albar965) [reply src/search/airportsearch.cpp:406]: I already fixed the restore with c9b649caee99a8e3b0b3d5704fb6548f31703d90
Alex

Comment (u-an-i) [reply src/search/airportsearch.cpp:406]: yes, i saw that. I think i have pulled that and built lnm with that. My built version still does not restore the values.

Comment (albar965) [reply src/search/airportsearch.cpp:406]: Will check.

**Comment by albar965** [comment]
FYI: Looking at `MainWindow::routeNewFromAirports()` since it deletes user made changes in the current plan and drops them from the undo stack. Also leaves two entries in the undo stack for departure and destination.
Everything else is works perfectly. :+1: 
Alex

**Comment by u-an-i** [comment]
Hi @albar965 ,

do you mean
image
?

I added this because of this comment:
image
`in search::airportsearch.cpp`

I always got the nagging dialog and it disrupted my experience.

Do you want me to do something ?

**Comment by albar965** [comment]
> Do you want me to do something ?

Nope. I'm about done changing this. The experience should be the same as when using the route description dialog window where no nagging appears at and all changes saved to the undo/redo stack. This avoids data loss for the user.

**Comment by albar965** [comment]
The idea is:
- Load a plan. 
- Change it.
- Start random generation.
- Accept flight. No nagging dialog.
- Undo #1 -> Back to changed flight from radom.
- Undo #2 -> Back to unchanged flight as loaded

**Comment by albar965** [comment]
5e758fd001e21f4bb8bf104a744fdace30f28237
Looking ok so far.
BTW: State of the random distance is saved here.

**Comment by u-an-i** [comment]
looks appropriate.

**Pull request merged**